File: ahma0/PGS_84512.java
import java.util.*;
import java.io.*;

class Solution {
    
    public int solution(String word) {

        Map<Character, Integer> map = new HashMap<>();
        
        map.put('A', 1);
        map.put('E', 2);
        map.put('I', 3);
        map.put('O', 4);
        map.put('U', 5);
        
        int answer = 0;
        
        for(int i = 0; i < word.length(); i++) {
            answer += (map.get(word.charAt(i)) - 1) * getPow(i) + 1;
        }

        return answer;
    }
    
    private int getPow(int index) {
        int cnt = 0;
        
        for(int i = 0; i <= 4 - index; i++) {
            cnt += Math.pow(5, i);
        }
        
        return cnt;
    }
}

File: chaeryeon823/PGS_84512.js
// 모음 사전
// lv 2
// 완전탐색
// 경우의 수

function solution(word) {
  let answer = 0;
  let cnt = [781, 156, 31, 6, 1];
  let arr = ["A", "E", "I", "O", "U"];
  for (let i = 0; i < word.length; i++) {
    answer += arr.indexOf(word[i]) * cnt[i] + 1;
  }
  return answer;
}
